### PR TITLE
Create Trip Details package

### PR DESCRIPTION
**Pull request by evansiroky**
Refs https://github.com/opentripplanner/otp-ui/issues/38

### CODE REVIEW

File: packages/trip-details/src/TripDetails.story.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { storiesOf } from "@storybook/react";
+import { withA11y } from "@storybook/addon-a11y";
+import { withInfo } from "@storybook/addon-info";
+import styled from "styled-components";
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/TripDetails.story.js]: Suggestion: Move `styled` import below React's.

File: packages/trip-details/src/index.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+};
+
+TripDetails.defaultProps = {
+  className: null,
+  longDateFormat: null,
+  routingType: "ITINERARY",
+  timeOptions: null
+};
+
+class TripDetail extends Component {
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/index.js]: Move this to its own file, e.g. `/TripDetails/index.js`

File: packages/trip-details/src/TripDetails.story.js (hunk 1 above)
Comment (evansiroky) [reply packages/trip-details/src/TripDetails.story.js]: I'm declining to do this since it would no longer be an alphabetic organization of imports.

**Comment by evansiroky** [comment]
Just pushed a commit that implements most of @binh-dam-ibigroup's suggestions.

File: packages/trip-details/src/index.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
+import {
+  calculateFares,
+  calculatePhysicalActivity
+} from "@opentripplanner/core-utils/lib/itinerary";
```
Comment (fpurcell) [inline packages/trip-details/src/index.js]: A note about using .../lib/... paths on import statements:

In a future PR, I'm going to go thru all the files in otp-ui and switch all the .../lib/... import paths to .../src/...  Cole pointed out that by us using transpiled .../lib/... code, we're preventing webpack from tree-shaking the otp-ui code, resulting in larger footprints than might be necessary.  We might do more, based on Cole's continued research in making the code performant.

We're not there yet tho, so no action necessary just yet. More just a heads-up.
We're at the 'get it running phase' right now.
Next, we'll move to 'get it running correctly'.
Then when (if) we get to the 'get it running fast' phase, expect said changes...

Comment (fpurcell) [reply packages/trip-details/src/index.js]: BTW, I've tried out the .../src/... paths in a dev bundle on npm (https://www.npmjs.com/package/@ott_test/transit-vehicle-overlay), and I was able to pull in base-map, core-utils and transit-vehicle-overlay, which used the .../src/... paths, and my simple_app (https://github.com/trimet-sandbox/simple_app), as well as dev version of trimet.org were able to pull in the code and do a latter transpile in building the app.  More testing necessary, but...

Comment (landonreed) [reply packages/trip-details/src/index.js]: Do you want to copy this comment to an issue so that we can track it outside of this PR, @fpurcell?

**Pull request merged**